lib/Partitioner/Partitioner.cpp

```diff
@@ -308,9 +360,18 @@ void Partitioner::initOpComputeTime(Function *F) {
       auto *FCN = llvm::dyn_cast<FullyConnectedNode>(&node);
       auto inputDims = FCN->getInput().dims();
       auto wtDims = FCN->getWeights().dims();
-      totalOps = 2 * inputDims[0] * inputDims[1] * wtDims[1];
+      totalOps = 2 * inputDims[0] * inputDims[1] * wtDims[0];
+      break;
+    }
+#ifdef GLOW_WITH_HABANA
+    case Kinded::Kind::HabanaFullyConnectedNodeKind: {
```

Comment:
Having backend-specific stuff in partitioner seems a bit entangled to me.
I see how this is backend-specific but it seems like we should factor this stuff out to maybe be part of the backend. We don't have have to do that in this pr but should look into it. Ideally most of these cases would be moved somewhere else I think.

Reply:
Agreed in general. This does need to be refactored at a later point. But I would want that to be in a different PR.

Reply:
Agree with @jackm321 . In fact, in partitioner, the backend-specific info will comes from configure file (passed through device configs).  In addition, I think the "roofline" calculation can be moved from Partitioner.cpp and added in some support file or utility file.

Reply:
I agree that some refactoring is needed here eventually but I'd really like to move the partitioner diffs forward without a ton of redesign since they're important to understanding/optimizing perf on devices.  We can tidy this up as things become more stable.

Reply:
Sounds good